### ISSUE TITLE
IllegalMonitorException/RuntimeException

### ISSUE THREAD

**Issue by danielgomezrico** [googlemaps/android-maps-utils#768]
#### Summary

Im getting a crash on Crashlytics related to the `ClusterManager`, so Im not really sure if this is related to the GMaps SDK itself or this one, should I move this to the maps tracker?

Im using the `MapView` inside a `Fragment`.

#### Versions
- Kotlin 1.3.72
- Android gradle cache: "com.android.tools.build:gradle:4.0.1"
- Google Services: com.google.gms:google-services:4.3.3
- "com.google.android.gms:play-services-maps:17.0.0"
- "com.google.maps.android:maps-ktx:1.8.0"
- "com.google.maps.android:maps-utils-ktx:1.8.0"

#### Device
```
Device
Brand: Xiaomi
Model: Mi A2
Orientation: Portrait
RAM free: 2.23 GB
Disk free: 56.6 GB
Operating System
Version: 10
Orientation: Portrait
Rooted: No
```

#### Stacktrace
```java
Fatal Exception: java.lang.RuntimeException: An error occurred while executing doInBackground()
       at android.os.AsyncTask$4.done(AsyncTask.java:399)
       at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
       at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
       at java.util.concurrent.FutureTask.run(FutureTask.java:271)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:919)

Caused by java.lang.IllegalMonitorStateException
       at java.util.concurrent.locks.ReentrantReadWriteLock$Sync.tryRelease(ReentrantReadWriteLock.java:373)
       at java.util.concurrent.locks.AbstractQueuedSynchronizer.release(AbstractQueuedSynchronizer.java:1291)
       at java.util.concurrent.locks.ReentrantReadWriteLock$WriteLock.unlock(ReentrantReadWriteLock.java:1138)
       at com.google.maps.android.clustering.algo.AbstractAlgorithm.unlock(AbstractAlgorithm.java:37)
       at com.google.maps.android.clustering.ClusterManager$ClusterTask.doInBackground(ClusterManager.java:301)
       at com.google.maps.android.clustering.ClusterManager$ClusterTask.doInBackground(ClusterManager.java:294)
       at android.os.AsyncTask$3.call(AsyncTask.java:378)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:919)
```

#### Expected behavior

No crash

**Comment by arriolac** [comment]
This was fixed in #761, please update to the latest version (2.0.3) to receive the fix. Let us know if you are still seeing it after updating the version

**Comment by danielgomezrico** [comment]
I will update to try it out and come back with news, thanks the happy response! 👍
